administrator/components/com_content/Helper/ContentHelper.php

```diff
@@ -104,36 +134,38 @@ public static function countItems(&$items)
 
 		foreach ($items as $item)
 		{
-			$item->count_trashed = 0;
-			$item->count_archived = 0;
+			$item->count_trashed     = 0;
 			$item->count_unpublished = 0;
-			$item->count_published = 0;
-			$query = $db->getQuery(true);
-			$query->select('state, count(*) AS count')
-				->from($db->qn('#__content'))
-				->where('catid = ' . (int) $item->id)
-				->group('state');
-			$db->setQuery($query);
-			$articles = $db->loadObjectList();
+			$item->count_published   = 0;
+
+			$query  = $db->getQuery(true);
+
+			$query	->select($db->qn('condition'))
+					->select('COUNT(*) AS ' . $db->qn('count'))
+					->from($db->qn('#__content', 'c'))
+					->from($db->qn('#__workflow_states', 's'))
+					->from($db->qn('#__workflow_associations', 'a'))
+					->where($db->qn('a.item_id') . ' = ' . $db->qn('c.id'))
+					->where($db->qn('s.id') . ' = ' . $db->qn('a.state_id'))
+					->where('catid = ' . (int) $item->id)
+					->where('a.extension = ' . $db->quote('com_content'))
+					->group($db->qn('condition'));
+
+			$articles = $db->setQuery($query)->loadObjectList();
 
 			foreach ($articles as $article)
 			{
-				if ($article->state == 1)
+				if ($article->condition == 1)
 				{
 					$item->count_published = $article->count;
 				}
 
-				if ($article->state == 0)
+				if ($article->condition == 0)
 				{
 					$item->count_unpublished = $article->count;
 				}
 
-				if ($article->state == 2)
-				{
-					$item->count_archived = $article->count;
-				}
-
-				if ($article->state == -2)
+				if ($article->condition == -2)
 				{
 					$item->count_trashed = $article->count;
 				}
```

Comment:
This approach requires changes when states/conditions are added or removed. Better:
Define constants, eg.
```php
class WorkflowConditions
{
    const PUBLISHED   =  1;
    const UNPUBLISHED =  0;
    const ARCHIVED    =  2;
    const TRASHED     = -2;
}
```
The loop should look like
```php
$item->count = [];
foreach ($articles as $article)
{
    $item->count[$article->condition] = ($item->count[$article->condition] ?: 0) + $article->count;
}
```
In subsequent code, you'll access 
```php
$item->count[WorkflowConditions::PUBLISHED]
```
instead of
```php
$item->count_published
```